src/gui/owncloudgui.cpp

```diff
@@ -703,4 +727,19 @@ void ownCloudGui::slotRemoveDestroyedShareDialogs()
 }
 
 
+void ownCloudGui::slotDialogVisibilityChanged(bool visible)
+{
+    auto dialog = qobject_cast<QDialog *>(sender());
```

Comment:
Just a sidenote: with an event filter like I'm advocating we'd have the object and wouldn't need to rely on `sender()`.